elastica/memory_block/memory_block_rod.py

```diff
@@ -109,6 +165,29 @@ def __init__(self, systems: Sequence):
         _reset_scalar_ghost(self.rest_lengths, self.ghost_elems_idx, 1.0)
         _reset_scalar_ghost(self.rest_voronoi_lengths, self.ghost_voronoi_idx, 1.0)
 
+        # Compute strains for the block
+        _compute_sigma_kappa_for_blockstructure(self)
+
+        # If n_elems_with_boundary defined and passed with kwargs, then this rod is ring and we need to know
+        # how many boundary elements is rod containing.
+        if n_ring_rods != 0:
+            for sys_idx, system_to_be_added in enumerate(system_ring_rod):
+                if np.count_nonzero(system_to_be_added.rest_sigma) == 0:
+                    # Ring rod has to have non-zero rest sigma. If user did not set something, then Elastica will
+                    # calculate it.
+                    system_to_be_added.rest_sigma[:] = system_to_be_added.sigma[:]
+                if np.count_nonzero(system_to_be_added.rest_kappa) == 0:
+                    # Ring rod has to have non-zero rest kappa. If user did not set something, then Elastica will
+                    # calculate it.
+                    system_to_be_added.rest_kappa[:] = system_to_be_added.kappa[:]
+
+            _synchronize_periodic_boundary_of_vector_collection(
+                self.rest_sigma, self.periodic_boundary_elems_idx
+            )
+            _synchronize_periodic_boundary_of_vector_collection(
+                self.rest_kappa, self.periodic_boundary_voronoi_idx
+            )
```

Comment:
so I have some confusion, for ring rods, are the end nodes coinciding or the end elements? because here the elemental properties are being equated (actually elemental and voronoidal)

Reply:
resolved via 5857ae8